cardano_node_tests/tests/test_pools.py

```diff
@@ -106,12 +106,16 @@ def _check_pool_deregistration_in_db(stake_pool_id: str, retiring_epoch: int):
     return None
 
 
-def _check_pool_data_in_db(ledger_pool_data: dict, db_pool_data: dbsync_utils.PoolDataRecord):
+def _check_pool_data_in_db(  # noqa: C901
+    ledger_pool_data: dict, pool_id: str
+) -> Optional[dbsync_utils.PoolDataRecord]:
```

Comment:
I changed the function so it calls `dbsync_utils.get_pool_data` itself after checking that db-sync is available, instead of getting the data as an argument. This way we don't need to add check for db-sync availability to `dbsync_utils.get_pool_data` (separation of concerns) and have just few "entry points" where these checks are performed.

We now also return the pool information from the function (if db-sync is available), same way as we do in `check_tx` in https://github.com/input-output-hk/cardano-node-tests/blob/a4aedef379e4a238f0c283fc275e13c463a2bbe1/cardano_node_tests/utils/dbsync_utils.py#L775 - get the data, check it, return it.